geniushubclient/zone.py

```diff
@@ -33,15 +34,128 @@ def alphanum_key(k):
     return sorted(dict_list, key=alphanum_key)
 
 
+@dataclass
+class CurrentState:
+    """The current state of the zone.
+
+        Intended to be properties that can change in the lifecycle of the zone.
+    """
+
+    is_active: bool = False
+    is_requesting_heat: bool = False
+    mode: int = 0
+    mode_name: str = ""
+    temperature: float = 0.0
+    set_point: float = 0.0
+    is_occupied: bool = False
+
+
+@dataclass
+class Properties:
+    """The properties of the zone.
+
+        Intended to be the properties that are fixed for the lifecycle of the zone
+    """
+
+    type: int = 0
+    type_name: str = ""
+    has_room_sensor: bool = False
+
+
+@dataclass
+class DaySchedule:
+    """The day in a schedule. Contains a list of heating periods."""
+
+    index: int
+    name: str
+    default_setpoint: float
+    heating_periods: List
+
+
+@dataclass
+class HeatingPeriod:
+    """The heating period data."""
+
+    start: int
+    end: int
+    set_point: float
+
+
+@dataclass
+class Override:
+    """The override data for the zone."""
+
+    duration: int
+    set_point: float
+
+
+@dataclass
+class WarmUp:
+    """The warmup data for the zone."""
+
+    enabled: bool = False
+    calcs_enabled: bool = False
+    lag_time: int = 0
+    rise_rate: float = 0
+    rise_time: int = 0
+    total_time: int = 0
+
+
+@dataclass
+class Reactive:
+    """The reactive data for the zone."""
+
+    activity_level: float = 0.0
+
+
+@dataclass
+class Footprint:
+    """The footprint data for the zone."""
+
+    is_night: bool = False
+    reactive: Reactive = None
+
+
+@dataclass
+class Trigger:
+    """The trigger data for the zone."""
+
+    reactive: bool = False
+    output: int = 0
+
+
 class GeniusZone(GeniusBase):
     """The class for a Genius Zone."""
 
     def __init__(self, zone_id, raw_json, hub) -> None:
         super().__init__(zone_id, raw_json, hub, ATTRS_ZONE)
 
+        self._state = CurrentState()
+        self._properties = Properties()
+        self._footprint = Footprint()
+        self._trigger = Trigger()
+        self._timer_schedule = []
+        self._footprint_schedule = []
+        self._override = None
+        self._warmup = WarmUp()
         self.device_objs = []
         self.device_by_id = {}
 
+        if self._hub.api_version == 3:
+            self.update(raw_json)
+
+    def update(self, json):
+        """Parse the json and use it to populate the data classes."""
+        self.update_state(json)
+        self.update_properties(json)
+        self.update_timer_schedule(json)
+        self.update_footprint_schedule(json)
+        self.update_footprint(json)
+        self.update_warmup(json)
+        self.update_override(json)
+        self.update_trigger(json)
+        self.update_is_occupied(json)
+
     @property
     def data(self) -> Dict:
```

Comment:
I'm not sure why this has been left in - all the main JSON conversion code is now duplicated? We should be able to do something similar to:
https://github.com/zxdavb/geniushub-client/blob/b419463f6eba2436753777025faab8d7cf83f928/geniushubclient/zone.py#L167-L176

Reply:
This may be easier if we are careful with the dataclass' namespaces.

Reply:
I presume that you are referring to the data property? I left it in so that my changes were purely additive, I figured that as it's a very large change, it would be a lot safer to get the new stuff in, be happy that it works, then remove the old conversion code.

Happy to delete it if you are!

Reply:
Please take it out.

Reply:
See my reply in Gitter as to why I left this in.